server/lib/ga-collector.js

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var extend = require('extend');
+var ua = require('universal-analytics');
+var logger = require('mozlog')('server.ga');
+var config = require('./configuration');
+
+var PUBLIC_URL = config.get('public_url');
+var ANALYTICS_ID = config.get('google_analytics_id');
+var SIGNUP_FLOW = 'Firefox Accounts Sign-up Flow';
+
+// note: if you use labels (el), then a value must be present (ev).
+var GA_EVENTS = {
+  'screen.signup': {
+    ec: SIGNUP_FLOW,
+    ea: 'page load',
+    el: 'Sign Up Page',
+    ev: 1
+  },
+  'signup.success': {
+    ec: SIGNUP_FLOW,
+    ea: 'registered',
+    el: 'regular',
+    ev: 1
+  },
+  'oauth.signup.success': {
+    ec: SIGNUP_FLOW,
+    ea: 'registered',
+    el: 'oauth',
+    ev: 1
+  },
+  'verify-email.verification.success': {
+    ec: SIGNUP_FLOW,
+    ea: 'verified email address'
+  }
+};
+
+function GACollector(options) {
+  options = options || {};
+
+  this.analyticsId = options.analyticsId || ANALYTICS_ID;
+}
+
+GACollector.prototype = {
+  /**
+   * Send a formatted metrics object to Google Analytics
+   *
+   * @param {Object} body
+   */
+  write: function (body) {
+    if (! body || ! body.events || ! body.events.length > 0 || ! this.analyticsId) {
```

Comment:
Removing the 'google_analytics_id' from config is an easy kill switch if we need to disable GA reporting. Good.